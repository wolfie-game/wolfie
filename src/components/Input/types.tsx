export type InputProps = {
  styleName?: string
  type: 'text' | 'password'
  name: string
  value: string
  readOnly?: boolean
  placeholder?: string
  handler?: (event: any) => void
  error?: boolean
}
