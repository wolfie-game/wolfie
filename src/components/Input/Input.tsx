import React from 'react'
import { InputProps } from './types'

type Props = InputProps

const Input = (props: Props) => {
  return (
    <input 
      className={props.styleName} 
      type={props.type} 
      name={props.name} 
      value={props.value} 
      readOnly={props.readOnly} 
      placeholder={props.placeholder} 
      onChange={props.handler}
    />
  )
}

export default Input