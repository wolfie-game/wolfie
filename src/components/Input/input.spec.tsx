import React from 'react'
import Input from './Input'
import renderer from 'react-test-renderer'

it('render correctly Input component', () => {
  const InputComponent = renderer
    .create(<Input type={'text'} name={''} value={''} />)
    .toJSON()
  expect(InputComponent).toMatchSnapshot()
})
