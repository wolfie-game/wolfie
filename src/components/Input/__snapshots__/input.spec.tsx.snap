// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Input component 1`] = `
<div
  className="input__field"
>
  <label
    className="input__error"
  >
    
  </label>
  <input
    name=""
    onBlur={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`render correctly text component 1`] = `
<div
  className="input__field"
>
  <label
    className="input__error"
  >
    
  </label>
  <input
    name=""
    onBlur={[Function]}
    type="text"
    value=""
  />
</div>
`;
